tests/integration/test_charm.py

```diff
@@ -41,3 +42,26 @@ async def test_reachable(app: Application):
     response = requests.get(f"http://{ip_address}:{port}")
 
     assert response.status_code == 200
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_ingress_reachable(app: Application):
+    """
+    arrange: given charm has been built, deployed and related to a dependent application
+    act: when the dependent application is queried via the ingress
+    assert: then the response is HTTP 200 OK.
+    """
+    # Get the IP address which is in the status message
+    ip_regex = r"[0-9]+(?:\.[0-9]+){3}"
+    ip_address_match = re.findall(ip_regex, app.units[0].workload_status_message)
+    assert (
+        ip_address_match
+    ), f"could not find IP address in status message: {app.units[0].workload_status_message}"
+    ip_address = ip_address_match[0]
+    ps = subprocess.Popen(["echo", f"{ip_address} hello-kubecon"], stdout=subprocess.PIPE)  # nosec
+    subprocess.run(["sudo", "tee", "-a", "/etc/hosts"], stdin=ps.stdout)  # nosec
```

Comment:
Minor thing: Would be possible to call passing the host header instead of changing /etc/hosts like Indico?
https://github.com/canonical/indico-operator/blob/main/tests/integration/test_charm.py#L35

Reply:
for ingress specifically, it is necessary to change the /etc/hosts file to make the IP reachable and display the page correctly.